daemon/emer-daemon.c

```diff
@@ -211,15 +219,85 @@ is_uuid (GVariant *variant)
 }
 
 static void
-uuid_from_variant (GVariant *variant,
-                   uuid_t    uuid)
+buffer_event (EmerDaemon *self,
+              GVariant   *event)
+{
+  EmerDaemonPrivate *priv = emer_daemon_get_instance_private (self);
+
+  g_variant_ref_sink (event);
+  gsize size_with_type = get_size_with_type (event);
+  gsize new_bytes_buffered = priv->num_bytes_buffered + size_with_type;
+  if (new_bytes_buffered > priv->max_bytes_buffered ||
+      /* Don't get wedged by a large event. */
+      new_bytes_buffered > MAX_REQUEST_PAYLOAD)
+    {
+      g_variant_unref (event);
```

Comment:
Not sure; I worry it would flood the logs of users who are permanently offline if we logged every time the buffer was full. I could see having a warning for an egregiously large event, although it would be a lot more useful to have such a warning in the event recorder daemon where a programmer adding a new event might actually see it. What do you think?